File: core/src/main/java/com/example/submarine/Treasure.java
package com.example.submarine;

import com.badlogic.gdx.math.Rectangle;

public class Treasure {
    public float x;
    public float y;
    public float width;
    public float height;
    public boolean collected;

    public Treasure(float x, float y, float width, float height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
        this.collected = false;
    }

    public Rectangle getBounds() {
        return new Rectangle(x, y, width, height);
    }
}

File: core/src/main/java/com/example/submarine/Treasure.kt
package com.example.submarine

import com.badlogic.gdx.math.Rectangle

class Treasure(@JvmField var x: Float, @JvmField var y: Float, @JvmField var width: Float, @JvmField var height: Float) {
    @JvmField
    var collected: Boolean = false

    val bounds: Rectangle
        get() = Rectangle(x, y, width, height)
}
